default/python/AI/ShipDesignAI.py

```diff
@@ -852,10 +855,8 @@ def update_stats(self, ignore_species=False):
             elif partclass in ARMOUR:
                 self.structure += capacity
             elif partclass in WEAPONS:
-                if capacity in self.attacks:
-                    self.attacks[capacity] += 1
-                else:
-                    self.attacks[capacity] = 1
+                shots = self._calculate_weapon_shots(part, ignore_species=ignore_species)
+                self.attacks[capacity] = self.attacks.get(capacity, 0) + shots
```

Comment:
I am lost here.

We add number of shots to attack damage.
`self.attacks[capacity] += shots`